### PR TITLE
Display one image and slicer

### PR DESCRIPTION
**Pull request by mprostakk**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,6 @@
 venv/
-.idea/
\ No newline at end of file
+.idea/
+.vscode/settings.json
+dicom_app/__pycache__/__init__.cpython-37.pyc
```
Comment (mprostakk) [inline .gitignore]:
```suggestion
dicom_app/__pycache__/__init__.cpython-37.pyc
```
Maybe use **.pyc instead if specific folder, same for __pycache__ in whole project

File: dicom_app/dicom_reader.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+from pydicom.data import get_testdata_files
+
+from PIL import Image
+from PyQt5.QtGui import QImage
+import matplotlib.pyplot as plt
+
+
+def convert_dicom_to_QImage(cvImg):
+    height, width = cvImg.shape
+    bytesPerLine = 3 * width
```
Comment (mprostakk) [inline dicom_app/dicom_reader.py]: bytesPerLine => bytes_per_line
Don't use CamelCase, use snake_case. PyQT uses camelcase ... (weird)

File: dicom_app/dicom_reader.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import pydicom
+from pydicom.data import get_testdata_files
+
+from PIL import Image
+from PyQt5.QtGui import QImage
+import matplotlib.pyplot as plt
+
+
+def convert_dicom_to_QImage(cvImg):
```
Comment (mprostakk) [inline dicom_app/dicom_reader.py]:
```suggestion
def convert_dicom_to_QImage(cv_img) -> QImage:
```

File: gui/main_window.py [hunk 4]
```diff
@@ -14,16 +21,39 @@ class DicomViewer(QMainWindow):
 
     def __init__(self):
         super(DicomViewer, self).__init__()
+        self.test_dicom = dicom_reader.get_dicom()
+        self.initUI()
+
+    def initUI(self):
```
Comment (mprostakk) [inline gui/main_window.py]: use snake_case

File: gui/main_window.py [hunk 5]
```diff
@@ -14,16 +21,39 @@ class DicomViewer(QMainWindow):
...
+        # image
+        self.figure = Figure(figsize=(5, 3))
+        self.canvas = FigureCanvas(self.figure)
+        self.ax = self.figure.subplots()
+        self.ax.imshow(dicom_reader.get_image_from_dicom(self.test_dicom, 0))
+        self.ax.set_axis_off()
+        layout.addWidget(self.canvas)
+
+        # widget
+        widget = QWidget()
```
Comment (mprostakk) [inline gui/main_window.py:60]: widget doens't really mean anything, maybe something more specific?

File: gui/main_window.py [hunk 6]
```diff
@@ -32,6 +62,11 @@ def __init__(self):
         self.setMinimumWidth(800)
         self.setMinimumHeight(600)
 
+    def changeValue(self, value):
+        # print(str(value))
```
Comment (mprostakk) [inline gui/main_window.py]: Remove comments, prints for sure

File: .gitignore (hunk 1 above)
Comment (xszym) [reply .gitignore]: Right!

File: dicom_app/dicom_reader.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import pydicom
+from pydicom.data import get_testdata_files
+
+from PIL import Image
+from PyQt5.QtGui import QImage
+import matplotlib.pyplot as plt
```
Comment (NorbertOzga) [inline dicom_app/dicom_reader.py]: Please reorder imports to be consistent with PEP8

File: gui/main_window.py [hunk 8]
```diff
@@ -32,6 +62,9 @@ def __init__(self):
         self.setMinimumWidth(800)
         self.setMinimumHeight(600)
 
+    def change_value(self, value):
+        self.label.setText(str(value))
+
     def create_actions(self):
         self.test_action = QAction("&Test...", self, shortcut="Ctrl+T", triggered=self.test)
         self.action_exit = QAction("&Exit...", self, shortcut="Alt+F4", triggered=self.on_exit)
```
Comment (NorbertOzga) [inline gui/main_window.py]: Line too long (80 chars max)

File: gui/main_window.py [hunk 9]
```diff
@@ -1,9 +1,16 @@
 import sys
 import logging
+import numpy as np
+
 
 from PyQt5 import QtGui
-from PyQt5.QtWidgets import QAction, QLabel, QScrollArea, QMenu, QSizePolicy, QApplication, QMainWindow
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import QAction, QLabel, QScrollArea, QHBoxLayout, QMenu, QWidget, QSizePolicy, QApplication, QMainWindow, QSlider, QHBoxLayout
```
Comment (NorbertOzga) [inline gui/main_window.py]: Line too long, please use \

File: dicom_app/dicom_reader.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+
+def convert_dicom_to_QImage(cv_img):
+    height, width = cv_img.shape
+    bytes_per_line = 3 * width
```
Comment (NorbertOzga) [inline dicom_app/dicom_reader.py]: Magic number, at least, there should be comment explaining why width is multiply by 3

File: gui/main_window.py [hunk 11]
```diff
@@ -14,16 +21,39 @@ class DicomViewer(QMainWindow):
...
+    def initUI(self):
+
+        layout = QHBoxLayout()
+
+        # slider
+        sld = QSlider(Qt.Vertical, self)
+        sld.setGeometry(40, 30, 30, 200)
+        sld.setRange(0, 100) # len(self.test_dicom)
+        sld.valueChanged[int].connect(self.change_value)
+        layout.addWidget(sld)
```
Comment (mprostakk) [inline gui/main_window.py]: Don't use sld. Means nothing, use name slider or something more specific. Maybe use a functions to build widgets?

File: gui/main_window.py [hunk 12]
```diff
@@ -1,29 +1,29 @@
-import sys
 import logging
 
-from PyQt5 import QtGui
-from PyQt5.QtWidgets import QAction, QLabel, QScrollArea, QMenu, QSizePolicy, QApplication, QMainWindow
+from dicom_app import dicom_reader
```
Comment (NorbertOzga) [inline gui/main_window.py]: Local application imports should be last

Comment (NorbertOzga) [reply gui/main_window.py]: Following PEP8:

Imports should be grouped in the following order:

    1. Standard library imports.
    2. Related third party imports.
    3. Local application/library specific imports.

You should put a blank line between each group of imports.

File: dicom_app/dicom_reader.py [hunk 13]
```diff
@@ -0,0 +1,15 @@
+import pydicom
+
```
Comment (NorbertOzga) [inline dicom_app/dicom_reader.py]: Redundant blank line

File: main.py [hunk 14]
```diff
@@ -5,19 +5,9 @@
 from PyQt5 import QtWidgets
 import matplotlib.pyplot as plt
```
Comment (NorbertOzga) [inline main.py]:
```suggestion
import matplotlib.pyplot as plt
from PyQt5 import QtWidgets
```

File: dicom_app/dicom_reader.py [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+from pydicom.data import get_testdata_files
+
+
+def get_dicom(filename='CT_small.dcm'):
+    files = get_testdata_files(filename)
+    return files
+
+
+def get_image_from_dicom(dicon_files, index):
```
Comment (NorbertOzga) [inline dicom_app/dicom_reader.py]: typo dicon_files -> dicom_files

File: dicom_app/dicom_reader.py (hunk 2 above)
Comment (NorbertOzga) [reply dicom_app/dicom_reader.py]: More specifically it is mixedCase

**Pull request merged**